src/TokenHandler/CreditRegistry.mo

```diff
@@ -0,0 +1,101 @@
+import RBTree "mo:base/RBTree";
+import Principal "mo:base/Principal";
+import Time "mo:base/Time";
+import Option "mo:base/Option";
+import Iter "mo:base/Iter";
+
+import Journal "Journal";
+
+module {
+  public type StableData = (Int, [(Principal, Int)]);
+
+  /// Tracks credited funds (usable balance) associated with each principal.
+  public class CreditRegistry(
+    journal : Journal.Journal,
+    isFrozen : () -> Bool,
+  ) {
+    var map : RBTree.RBTree<Principal, Int> = RBTree.RBTree<Principal, Int>(Principal.compare);
```

Comment:
Suggestion: Make a class `Map` that wraps around RBTree. The `Map` class takes care of storing the 0 value efficiently (by removing the key from the RBTree). Then `CreditRegistry` does not have to care about implementation details of the map. CreditRegistry will become easier to read and we can later more easily swap out RBTree for a different data structure without having to touch CreditRegistry.